core/templates/pages/admin-page/misc-tab/admin-misc-tab.component.ts

```diff
@@ -0,0 +1,328 @@
+  explorationStatsRegenerationUtilityCsvRows: string[][] = [
+    [
+      'Exploration ID',
+      'No. of valid Exploration Stats',
+      'No. of valid State Stats',
+      'Missing Exploration Stats',
+      'Missing State Stats',
+      'Errors'
+    ]
+  ];
+  MAX_USERNAME_LENGTH: number = AppConstants.MAX_USERNAME_LENGTH;
+  regenerationMessage: string;
+  missingExplorationStatsRegenerationMessage: string;
+  csvUri: SafeResourceUrl;
+  csvFilename: string;
+  explorationIdsForRegeneratingMissingStats: string;
+  showDataExtractionQueryStatus: boolean;
+  dataExtractionQueryStatusMessage: string;
+  memoryCacheDataFetched: boolean;
+  oldUsername: string;
+  newUsername: string;
+  usernameToGrant: string;
+  usernameToRevoke: string;
+  userIdToGet: string;
+  userIdToDelete: string;
+  usernameToDelete: string;
+  expVersion: number;
+  stateName: string;
+  numAnswers: number;
+  expId: string;
+  topicIdForRegeneratingOpportunities: string;
+
+  constructor(
+    private windowRef: WindowRef,
+    private adminBackendApiService: AdminBackendApiService,
+    private adminTaskManagerService: AdminTaskManagerService,
+    private domSantizer: DomSanitizer
+  ) {}
+
+  resetExplorationStatsRegenerationUtilityCsvRows(): void {
+    this.explorationStatsRegenerationUtilityCsvRows.length = 1;
+  }
+
+  clearSearchIndex(): void {
+    if (this.adminTaskManagerService.isTaskRunning()) {
+      return;
+    }
+    if (!this.windowRef.nativeWindow.confirm(this.irreversibleActionMessage)) {
+      return;
+    }
+
+    this.setStatusMessage.emit('Clearing search index...');
+
+    this.adminTaskManagerService.startTask();
+    this.adminBackendApiService.clearSearchIndexAsync()
+      .then(() => {
+        this.setStatusMessage.emit('Index successfully cleared.');
+        this.adminTaskManagerService.finishTask();
+      }, errorResponse => {
+        this.setStatusMessage.emit('Server error: ' + errorResponse);
+        this.adminTaskManagerService.finishTask();
+      });
+  }
+
+  regenerateOpportunitiesRelatedToTopic(): void {
+    if (this.adminTaskManagerService.isTaskRunning()) {
+      return;
+    }
+    if (!this.windowRef.nativeWindow.confirm(this.irreversibleActionMessage)) {
+      return;
+    }
+    this.regenerationMessage = 'Regenerating opportunities...';
+    this.adminBackendApiService.regenerateOpportunitiesRelatedToTopicAsync(
+      this.topicIdForRegeneratingOpportunities).then(response => {
+      this.regenerationMessage = (
+        'No. of opportunities model created: ' +
+        response);
+    }, errorResponse => {
+      this.regenerationMessage = ('Server error: ' + errorResponse);
+    });
+  }
+
+  populateExplorationStatsRegenerationCsvResult(expIds: string[]): void {
+    if (expIds.length === 0) {
+      let csvContent = this.explorationStatsRegenerationUtilityCsvRows.map(
+        row => row.join(',')).join('\n');
+      this.csvUri = this.domSantizer.bypassSecurityTrustResourceUrl(
```

Comment:
This function is equivalent of `$sce.trustAsResourceUrl` which was used earlier.
If there is a better alternative, please suggest.

Reply:
Could you check with @kevintab95 whether we could remove this functionality altogether? Would prefer to limit how much we do such bypassing.

Reply:
Yup confirming that this can be removed. Thanks!

Reply:
done